source/faas-automated-testing/faas-automated-testing.rst

```diff
@@ -0,0 +1,303 @@
++-------------------------------+-------------------------------------+
+| DRIVERS_ATLAS_PUBLIC_API_KEY  | The Atlas public API key            |
++-------------------------------+-------------------------------------+
+| DRIVERS_ATLAS_PRIVATE_API_KEY | The Atlas private API key           |
++-------------------------------+-------------------------------------+
+| DRIVERS_ATLAS_LAMBDA_USER     | The Atlas cluster user name         |
++-------------------------------+-------------------------------------+
+| DRIVERS_ATLAS_LAMBDA_PASSWORD | The Atlas cluster user password     |
++-------------------------------+-------------------------------------+
+| DRIVERS_ATLAS_GROUP_ID        | The driver's Atlas group id         |
++-------------------------------+-------------------------------------+
+| LAMBDA_STACK_NAME             | The driver's Lambda stack name      |
++-------------------------------+-------------------------------------+
+| AWS_REGION                    | The function AWS region             |
++-------------------------------+-------------------------------------+
+| AWS_ACCESS_KEY_ID             | Assume role atuomatically sets this |
++-------------------------------+-------------------------------------+
+| AWS_SECRET_ACCESS_KEY         | Assume role automatically sets this |
++-------------------------------+-------------------------------------+
+| AWS_SESSION_TOKEN             | Assume role automatically sets this |
++-------------------------------+-------------------------------------+
+
+
+This is an example function in the Evergreen config that accomplishes this, using
+subprocess.exec to execute a script that calls the drivers-evergreen-tools
+function inside of it:
+
+.. code:: yaml
+
+  run deployed aws lambda tests:
+  - command: ec2.assume_role
+    params:
+      role_arn: ${LAMBDA_AWS_ROLE_ARN}
+  - command: subprocess.exec
+    params:
+      working_dir: src
+      binary: bash
+      args:
+        - .evergreen/run-deployed-lambda-aws-tests.sh
+      env:
+        TEST_LAMBDA_DIRECTORY: ${PROJECT_DIRECTORY}/test/lambda
+        DRIVERS_TOOLS: ${DRIVERS_TOOLS}
+        DRIVERS_ATLAS_PUBLIC_API_KEY: ${DRIVERS_ATLAS_PUBLIC_API_KEY}
+        DRIVERS_ATLAS_PRIVATE_API_KEY: ${DRIVERS_ATLAS_PRIVATE_API_KEY}
+        DRIVERS_ATLAS_LAMBDA_USER: ${DRIVERS_ATLAS_LAMBDA_USER}
+        DRIVERS_ATLAS_LAMBDA_PASSWORD: ${DRIVERS_ATLAS_LAMBDA_PASSWORD}
+        DRIVERS_ATLAS_GROUP_ID: ${DRIVERS_ATLAS_GROUP_ID}
+        LAMBDA_STACK_NAME: dbx-node-lambda
+        AWS_REGION: us-east-1
+        AWS_ACCESS_KEY_ID: ${AWS_ACCESS_KEY_ID}
+        AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY}
+        AWS_SESSION_TOKEN: ${AWS_SESSION_TOKEN}
+
+Drivers MUST run the function on a single variant in Evergreen, in order to not
+potentially hit the Atlas API rate limit. The variant itself MUST be either a
+RHEL7/8 or Ubuntu 18/20 variant in order to have the SAM CLI installed.
```

Comment:
Ubuntu 18 is EOL and is being retired from EG. We should recommend RHEL7/8 or Ubuntu 20.

Reply:
Updated.